File: 백준/Bronze/29725. 체스 초보 브실이/체스 초보 브실이.cs
using System;

class Chess {
    static void Main() {
        var chess = new System.Collections.Generic.Dictionary<char, int> {
            {'K', 0}, {'k', 0},
            {'P', 1}, {'p', 1},
            {'N', 3}, {'n', 3},
            {'B', 3}, {'b', 3},
            {'R', 5}, {'r', 5},
            {'Q', 9}, {'q', 9}
        };

        int white = 0, black = 0;

        for (int i = 0; i < 8; i++) {
            string pieces = Console.ReadLine();
            foreach (char p in pieces) {
                if (p == '.') continue;

                if (char.IsUpper(p)) {
                    white += chess[p];
                } else {
                    black += chess[p];
                }
            }
        }

        Console.WriteLine(white - black);
    }
}

File: 백준/Bronze/29725. 체스 초보 브실이/체스 초보 브실이.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Map<Character, Integer> chess = new HashMap<>();
        chess.put('K', 0); chess.put('k', 0);
        chess.put('P', 1); chess.put('p', 1);
        chess.put('N', 3); chess.put('n', 3);
        chess.put('B', 3); chess.put('b', 3);
        chess.put('R', 5); chess.put('r', 5);
        chess.put('Q', 9); chess.put('q', 9);

        int white = 0, black = 0;
        Scanner sc = new Scanner(System.in);

        for (int i = 0; i < 8; i++) {
            String pieces = sc.nextLine();
            for (char p : pieces.toCharArray()) {
                if (p == '.') continue;

                if (Character.isUpperCase(p)) {
                    white += chess.get(p);
                } else {
                    black += chess.get(p);
                }
            }
        }

        System.out.println(white - black);
    }
}
